### ISSUE TITLE
non-unique labels

### ISSUE THREAD

**Issue by gergness** [tidyverse/haven#424]
Similar but not the same as #364 - it's currently possible to have multiple values mapped to a label with the same text.

```r
library(haven)
library(tibble)

x <- labelled(1:2, c("label" = 1, "label" = 2))
x
#> <Labelled integer>
#> [1] 1 2
#> 
#> Labels:
#>  value label
#>      1 label
#>      2 label
```

However, if you do this you can't assign to factors using option `levels = "label"`

```r
as_factor(x, "labels")
#> Error in `levels<-`(`*tmp*`, value = as.character(levels)): factor level [2] is duplicated
```

This is especially problematic because #390 used this in the print method.
```r
tibble(x = x)
#> Error in `levels<-`(`*tmp*`, value = as.character(levels)): factor level [2] is duplicated
```

Stata at least considers this data valid ([test.zip](https://github.com/tidyverse/haven/files/2842301/test.zip) attached), so I think the best solution is to fix the `as_factor()` code, but if not, I think it's important to at least change the tibble print method.

```r
read_dta("~/test.dta")
#> Error in `levels<-`(`*tmp*`, value = as.character(levels)): factor level [2] is duplicated
```

**Comment by briatte** [comment]
This issue makes it currently impossible to work with many Stata datasets using `haven`, since it kills the `print` function.

**Comment by gergness** [comment]
FWIW as an interim fix, you can set `options(haven.show_pillar_labels = FALSE)` to use the old print method.

**Comment by courtiol** [comment]
@elbersb the issue has been solved (actually at least twice: #425, #476) in pull requests, you now need to wait for the package maintainers to include the change in the package.

**Comment by gorcha** [comment]
Hi @hadley, can this fix be merged some time soon? For those of us who use haven regularly this essentially breaks dplyr printing (which is incredibly useful) for a lot of datasets.

It seems like a simple and non-controversial fix.

**Comment by hadley** [comment]
I will try and take a look in the near future but it’s likely to be 2-3 weeks. In the mean time, you can just install the version from the PR with the fix: `install_github(“tidyverse/haven#425”)`

**Comment by gorcha** [comment]
Thanks!
